code/modules/clothing/under/accessories.dm

```diff
@@ -110,6 +111,10 @@
 					else
 						user.visible_message("[user] pins \the [src] on [M]'s chest.", \
 											 "<span class='notice'>You pin \the [src] on [M]'s chest.</span>")
+						var/input = stripped_input(user,"Please input a reason for this commendation, it will be recorded by Nanotrasen.", ,"", 50)
+						if(input && !commended)
+							GLOB.commendations += "<b>[M.name]</b> was awarded the <font color='blue'>[name]</font>! \n- [input]"
```

Comment:
```M.real_name``` is probably more correct.
Otherwise at round end you'll get "John Doe (as Urist McShitter) was awarded..." and "Unknown was awarded..."